# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1291856131 18000
#      Wed Dec 08 19:55:31 2010 -0500
# Node ID 56c8a00a269f775ce6da5cfb9ac9f626de86da9e
# Parent  e773d57de57252789acfc562302e901059391966
fill.m: Allow multiple filled polygons.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,10 +1,11 @@
 2010-12-08  Ben Abbott <bpabbott@mac.com>
 
+	* plot/fill.m: Fix bug that implies nextplot = "add".
 	* image/imshow.m: Flip y-axis.
 
 2010-12-08  John W. Eaton  <jwe@octave.org>
 
 	* strings/strmatch.m: Avoid passing length of zero to strncmp.
 	Bug #31774.
 
 2010-12-07  John W. Eaton  <jwe@octave.org>
diff --git a/scripts/plot/fill.m b/scripts/plot/fill.m
--- a/scripts/plot/fill.m
+++ b/scripts/plot/fill.m
@@ -32,20 +32,20 @@ function retval = fill (varargin)
   htmp = [];
   iargs = __find_patches__ (varargin{:});
 
   oldh = gca ();
   unwind_protect
     axes (h);
 
     nextplot = get (h, "nextplot");
-    if (strncmp (nextplot, "replace", 7))
-      set (h, "nextplot", "add");
-    endif
     for i = 1 : length (iargs)
+      if (i > 1 && strncmp (nextplot, "replace", 7))
+        set (h, "nextplot", "add");
+      endif
       if (i == length (iargs))
         args = varargin (iargs(i):end);
       else
         args = varargin (iargs(i):iargs(i+1)-1);
       endif
       newplot ();
       [tmp, fail] = __patch__ (h, args{:});
       if (fail)
